File: user_text_util.py
import csv
import os
dict_list=[]
def reat_csv():
    """Открывает .csv файл и считывает 
    id(int) и name(str) построчно
    в лист словаря"""
    list=[]; arr=[]; f_csv= 'people.csv'
    with open(f_csv, 'r', newline='') as csvfile:
        reader = csv.reader(csvfile)
        for row in reader:
            arr.append(', '.join(row))
    for i in arr:
        dict_csv={};
        dict_csv["namber"]=int(i.split(';')[0])
        dict_csv["name"]=i.split(';')[1]
        list.append(dict_csv)    
    return list
def read_texts():
    """Открывает папку texts/ и считывает 
    вложенное в лист словаря"""
    path='texts/'
    dict_list=reat_csv()
    for rootdir, dirs, files in os.walk(path):
        for i in range(len(dict_list)):
            arr_line_mass=[]
            for file in files:  
                a=dict_list[i].get('namber')                      
                if int((file.split('-')[0]))==a:                  
                    with open(path+file, "r") as txtfile:
                        arr_line=[];
                        line=txtfile.read()
                        arr_line.append(line)
                    arr_line_mass.append(line)
            dict_list[i].update({"texts":arr_line_mass})
    return dict_list

def countAverageLineCount():
    """Функция находит и выводит среднее количество 
    строк на одного пользователя"""   
    global dict_list
    dict_list=  read_texts()       
    for i in range(len(dict_list)):
        ct=0
        a=dict_list[i].get('texts')
        for m in a:
            """Перебор строк в файле"""  
            ct=ct+1
            ct+=m.count('\n')
        dict_list[i].update({"count_line":round(ct/len(a))}) 

def replaceDates():
    """Функция находит "/" и заменяет на "-", записывает эти 
    данные в папку output_texts/, выводит количество замен"""  
    global dict_list
    dict_list=  read_texts()
    for i in range(len(dict_list)):
        ct=0; mass=[]
        a=dict_list[i].get('texts')            
        for m in a:
            """Перебор строк в файле"""  
            m1 = m.replace('/', '-') 
            ct+=m.count('/')
            mass.append(m1)
        dict_list[i].update({"texts":mass,"count_replace":ct})  
    output_texts(dict_list) 
def output_texts(dict_list):
    """Открывает папку output_texts/ и добавляет 
    туда изменённые файлы из папки texts/"""    
    path="output_texts/"
    for i in range(len(dict_list)): 
        for j in range(len(dict_list[i].get("texts"))):
            name_file=path+"{0}-{1:03d}.txt".format(i+1,j+1)
            b = dict_list[i].get("texts")[j]
            with open(name_file, 'w') as f:
                f.write(b)

def print_symbol(x,symbol,y):
    """Форма строки (имя,символ,число)"""
    print(x+symbol+y)
def print_full(s): 
    """Вывод строки со знаком (,) или (;)""" 
    global dict_list  
    for i in range(len(dict_list)): 
        a=str(dict_list[i].get("name"))
        b=str(dict_list[i].get("count_replace"))
        c=str(dict_list[i].get("count_line"))
        if b != 'None':            
            print_symbol(a,s,b)
        if c != 'None':        
            print_symbol(a,s,c)   
def comma():
    """Вывод строки со знаком (,)""" 
    print_full(',')
def semicolon ():
    """Вывод строки со знаком (;)""" 
    print_full(';')




replaceDates()
comma()


#countAverageLineCount()
#semicolon ()
# 
# #print(dict_list)

File: user_text_util.php
<pre>
<?php 
    echo "Hello,  kek <br>\n ";

echo '<pre>';

function reat_csv(){
    // Открывает .csv файл и считывает 
    // id(int) и name(str) построчно
    // в лист словаря
    // CSV файл в массив:
    global $arr;$arr=[];
    //$f_csv= 'people.csv'
    $lines = file('people.csv');    
    foreach($lines as $line)
    {   
        $arr_i=[];
        echo($line);        
        $index_i=explode(";", $line)[0];
        $name_i=explode(";", $line)[1];
        $arr_i+=['index'=>$index_i];
        $arr_i+=['name'=>$name_i];
        array_push($arr, $arr_i);
        // var_dump($arr);
        echo '<hr>';
    }
}


function read_texts()
{
    /*     Открывает папку texts/ и считывает 
    вложенные строки в лист массива */
    global $arr;
    reat_csv();
    echo "<br>\n";
    $a = [];
    $dir = './texts/';
    if ($handle = opendir($dir)) {
        while (false !== ($entry = readdir($handle))) {
            if ($entry != "." && $entry != "..") {
                $filename1 = $dir . $entry;
                $ffile3 = fopen($filename1, 'r');
                $contens = fread($ffile3, filesize($filename1));
                fclose($ffile3);
                $a+=[$entry=>$contens];
            }            
        }
        closedir($handle);
    }
    // var_dump($a);
    for ($i = 1; $i <= count($arr); $i++) {
        $b = [];
        foreach ($a as $key => $value) {
            $mykey = explode("-", $key)[0];
            if ($mykey==$i ){
                array_push($b, $value);
            }
        }
        $arr[$i-1]+=['texts'=>$b];
    }
}


function countAverageLineCount()
{
    /* Функция находит и выводит среднее количество 
    строк на одного пользователя */
    global $arr;
    read_texts();
    for ($i = 1; $i <= count($arr); $i++) {

        foreach ($arr[$i - 1] as $key => $value) {
            $c_str = 0;
            $mykey = $key;
            if ($mykey == 'texts') {
                // var_dump($value);
                foreach ($value as $v) {
                    // echo '<br>Количество строк' . count(explode("\n", $v)) . '<br>в ' . $v;
                    $c_str += count(explode("\n", $v));
                    
                }
                // echo count($value);
                $arr[$i - 1] += ['count_line' => round($c_str/count($value))];
            }

        }
    }
}
function replaceDates()
{
    /* Функция находит и выводит среднее количество 
    строк на одного пользователя */
    global $arr;
    read_texts();
    for ($i = 1; $i <= count($arr); $i++) {

        foreach ($arr[$i - 1] as $key => $value) {
            $c_str = 0;
            $mykey = $key;
            if ($mykey == 'texts') {
                // var_dump($value);
                foreach ($value as $v) {
                    // echo '<br>Количество строк' . count(explode("\n", $v)) . '<br>в ' . $v;
                    $c_str += substr_count( $v,'/');
                    echo $v;
                }
                // echo count($value);
                $arr[$i - 1] += ['count_replace' => $c_str];
            }

        }
    }
}
function print_symbol($x,$symbol,$y){
    /* Форма строки (имя,символ,число) */
    echo $x.$symbol.$y;
    echo '<hr>';
}
function print_full($s)
{
    /* Вывод строки со знаком (,) или (;)  */
    global $arr;
    for ($i = 1; $i <= count($arr); $i++) {
        $a = 0;$b = 0;$c = 0;
        foreach ($arr[$i - 1] as $key => $value) {
            $mykey = $key;
            if ($mykey == 'name') {
                $a = $value;
                // echo $a;
            }
            if ($mykey == 'count_replace') {
                $b = $value;
                // echo $b;
                var_dump($b);
            }
            if ($mykey == 'count_line') {
                $c = $value;
                // echo $c;
            }
            
        }   
        if ($b !== 0) {
            print_symbol($a, $s, $b);
        }
        if ($c !== 0) {
            print_symbol($a, $s, $c);
        }
    }
}
function comma(){
    /* Вывод строки со знаком (,)  */
    print_full(',');
}
function semicolon(){
    /* Вывод строки со знаком (;)  */
    print_full(';');
}

countAverageLineCount();
comma();
var_dump($arr);
